### ISSUE TITLE
Broken base64 brake contact images

### ISSUE THREAD

**Issue by sedrubal** [owncloud/contacts#315]
### Steps to reproduce
1. Addressbook with jpeg and png contact pictures (added by Android/DavDroid)
2. Open `/index.php/apps/contacts/`
### Expected behaviour

All contact pictures should be show
### Actual behaviour

Some contact pictures look like this:

![ocbug](https://cloud.githubusercontent.com/assets/5571650/14122419/04a6c1f0-f5fc-11e5-9e92-f3f89be34966.png)

The problem is quite similar to #175 .

The displayed src is 

``` html
src="data:image/png;base64,dataimage/jpegbase64/9j/4AAQSkZJRgABAQAAAQABAAD/2wB[...]"
```

changing it manually to 

``` html
src="data:image/jpeg;base64,/9j/4AAQSkZJRgABAQAAAQABAAD/
```

solves the problem. I think this only happens with `jpeg` images, because all images start with `data:image/png;` and if the image is `jpeg` `dataimage/jpegbase64` is wrongly appended.
### Server configuration

**Operating system**: `ubuntu:14.04`

**Web server:** `apache 2.4.7`

**Database:** `mariadb`

**PHP version:** `php 5.5.9`

**ownCloud version:** `9.0.0.19`

**Contacts version:** `1.0.0.0`

**Updated from an older ownCloud or fresh install:** `yes, all previous since oc 8.0?`

**Signing status (ownCloud 9.0 and above):** `No errors have been found.`

**List of activated apps:**

```
Enabled:
  - activity: 2.2.1
  - calendar: 1.0
  - comments: 0.2
  - contacts: 1.0.0.0
  - dav: 0.1.5
  - encryption: 1.2.0
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_external: 0.5.2
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - gallery: 14.5.0
  - news: 7.1.2
  - notes: 2.0.0
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - tasks: 0.8.1
  - templateeditor: 0.1
  - updatenotification: 0.1.0
```

**The content of config/config.php:** `...`

**Are you using external storage, if yes which one:** `no`

**Are you using encryption:** `no`

**Are you using an external user-backend, if yes which one:** `no`
### Client configuration

**Browser:** `Firefox 45.0.1`

**Operating system:** `fedora:23`
#### Web server error log

```
-
```
#### ownCloud log (data/owncloud.log)

```
-
```
#### Browser log

```
22:25:11.523 Content Security Policy: The page's settings blocked the loading of a resource at self ("script-src https://<url> 'unsafe-eval'"). contacts:1:0
22:25:11.611 Using //@ to indicate sourceMappingURL pragmas is deprecated. Use //# instead jquery.min.js:1:0
22:25:11.883 Content Security Policy: The page's settings blocked the loading of a resource at self ("script-src https://<url> 'unsafe-eval'"). contacts
22:25:12.213 mutating the [[Prototype]] of an object will cause your code to run very slowly; instead create the object with the correct initial [[Prototype]] value using Object.create dav.js:3072:415
22:25:13.320 Content Security Policy: The page's settings blocked the loading of a resource at http://remote.php/dav/addressbooks ("connect-src https://<urll>").
```

**Comment by DeepDiver1975** [comment]
Can we have an example vcard? THX

**Comment by sedrubal** [comment]
Sorry, I've no idea, how I can export single contacts as vCard. When I export them from my client software (Android contacts or Kontact (part of KDE)) there is no profile picture for these contacts...

**Comment by skjnldsv** [comment]
https://DOMAIN.EXT/remote.php/dav/addressbooks/users/USER/ADDRESSBOOK/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx.vcf

Replace the information with your data and replace the uuid by the contact you want to download.

**Comment by DeepDiver1975** [comment]
... or you just grab it from the database

**Comment by sedrubal** [comment]
Ah it took a while until I realized that the UID is part of the URL after clicking on the contact... Why isn't there an export button (e.g. next to trash icon in contacts app)?

I grabbed this by searching in the database. I've made the data anonymous... https://gist.github.com/sedrubal/b596ce247982a1aaf542a5a911230564#file-debug-vcard

The error might be this:

``` vcard
[...]
PHOTO:dataimage/jpegbase64/9j/4AAQSkZJRg[...]
[...]
```

instead of

``` vcard
[...]
PHOTO;ENCODING=b;TYPE=jpeg:/9j/4AAQSkZJRg[...]
[...]
```

This might be either a fail of a old owncloud version (I think this was in `oc 6` age :smile: ) or a old DavDroid fail...

Btw. accessing the contact via `HTTP` (`https://EXAMPLE.COM/remote.php/dav/addressbooks/users/USER/ADDRESSBOOK/20140129T234653.1006b69f26@192.168.0.101.vcf`) returned following error:

``` xml
<d:error><s:exception>Sabre\DAV\Exception\NotFound</s:exception><s:message>Card not found</s:message></d:error>
```

**Comment by gvde** [comment]
This is a bug of the old contacts app with photos in 4.0 vcards, i.e. you had this problem before upgrading to owncloud 9.0. See https://github.com/owncloudarchive/contacts/issues/1072

The problem is the broken uri. It should be
`PHOTO:data:image/jpeg;base64,/9j/`
instead of
`PHOTO:dataimage/jpegbase64/9j/`

Of course, it should be 4.0 card for that. Your card, however, is 3.0 and has an old style UID, thus either you have imported that contact a long, long time ago or something else messed up and provided a 4.0 style PHOTO data URI in a 3.0 card which wouldn't work either...

**Comment by skjnldsv** [comment]
Doesn't the vcard lib allow for reparations with the forgiving parameter?
http://sabre.io/vobject/vcard/

Do we use that? It could be great to have some kind of "debug" explaining your vcard is corrupted but we tried and imported it anyway?

**Comment by gvde** [comment]
Well, the broken PHOTO data uri is a result of the repair function in the old contacts app...

I would always recommend to import only what complies with the RFC. If something is broken tell the user where and what so that the user can manually fix it.

Automated repair should only be an additional option and I would even like to see that an import with automated repair provides a detailed comparison of what has been changed so that you can see whether the repair was successful or not. Because nothing is more annoying then importing a bunch of cards and after a while you find out that some cards are broken but you haven't noticed before...

**Comment by skjnldsv** [comment]
I agree!

**Comment by raimund-schluessler** [comment]
Try the fix in #422. This fixed the issue for me.

**Comment by DeepDiver1975** [comment]
> Try the fix in #422. This fixed the issue for me.

-> close
